osu.Game/Overlays/BeatmapListing/BeatmapListingPager.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public class BeatmapListingPager
+    {
+        private readonly IAPIProvider api;
+        private readonly RulesetStore rulesets;
+        private readonly string query;
+        private readonly RulesetInfo ruleset;
+        private readonly SearchCategory searchCategory;
+        private readonly SortCriteria sortCriteria;
+        private readonly SortDirection sortDirection;
+
+        public event PageFetchHandler PageFetched;
+        private SearchBeatmapSetsRequest getSetsRequest;
+        private SearchBeatmapSetsResponse lastResponse;
+
+        /// <summary> Reports end of results </summary>
+        private bool isLastPageFetched = false;
+        /// <summary> Job in process lock flag </summary>
```

Comment:
This does not parse to me as an English sentence.

Reply:
It does to me - C/++ terminology took liberty here I think, sorry
To removal anyway so doesn't matter